docs/Lexicon.rst

```diff
@@ -110,6 +116,20 @@ source code, tests, and commit messages. See also the `LLVM lexicon`_.
     Describes a type or function where making changes will break binary
     compatibility. See :doc:`LibraryEvolution.rst <LibraryEvolution>`.
 
+  generic environment
+    Provides context for interpreting a type that may have generic parameters
+    in it. Generic parameter types are normally just represented as "first
+    generic parameter in the outermost context" (or similar), so it's up to the
```

Comment:
You use the phrase "outermost context" without defining it. You should be more explicit.